test/unit/test_sign.py

```diff
@@ -75,10 +75,12 @@ def test_sct_verify_keyring_lookup_error(signer, monkeypatch):
 
     with pytest.raises(
         InvalidSCTError,
-        match="Invalid key ID in SCT: not found in current keyring",
-    ):
+    ) as excinfo:
         signer.sign(payload, token)
 
+    # The exception subclass is the one we expect.
+    assert isinstance(excinfo.value, InvalidSCTKeyError)
```

Comment:
can we check for this specific subclass with `pytest.raises`?

Reply:
We can, but I actually left it that way on purpose -- checking for the parent class being raised means that we haven't broken our exception contract around what this API raises. The `isinstance` check is then just a stand-in for ensuring that we got the right sub-error 🙂